src/InventoryManager1L.java

```diff
@@ -0,0 +1,210 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Kernel implementation of the InventoryManager component.
+ *
+ * Representation: - A Map<String, Integer> inventory, where keys are unique
+ * item names and values are positive integers representing item counts.
+ *
+ * Convention: - All item names (keys) in the map are non-null and non-empty. -
+ * All item quantities (values) are strictly positive integers (> 0).
+ *
+ * Correspondence: - For any item 'k', this.inventory.get(k) represents the
+ * number of units of item 'k' in the inventory.
+ */
+public class InventoryManager1L extends InventoryManagerSecondary {
+
+    private  Map<String, Integer> inventory;
+
+    /**
+     * Creates and returns a new empty representation for the inventory.
+     *
+     * @return a new empty map
+     */
+    private Map<String, Integer> createNewRep() {
+        return new HashMap<>();
+    }
+
+    /**
+     * No-argument constructor. Initializes the inventory as an empty map.
+     */
+    public InventoryManager1L() {
+        this.inventory = this.createNewRep();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Creates and returns a new, empty instance of InventoryManager.
+     *
+     * @return new instance of InventoryManager
+     */
+    @Override
+    public InventoryManager newInstance() {
+        return new InventoryManager1L();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Clears all items from the inventory.
+     */
+    @Override
+    public void clear() {
+        this.inventory = this.createNewRep();
```

Comment:
Usually, createNewRep is a void method. It would be called to effectively clear the data structure.